src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -2524,8 +2524,8 @@ int wm_vuldet_check_generic_package(sqlite3 *dbCVE,
 
     sqlite3_bind_text(stmt, 1, temp_name, -1, NULL);
     if (app_package_mac) {
-        /* Selecting all vulnerabilities for a mac app package if vendor is null */
-        if (!pkg_vendor) {
+        /* Selecting all vulnerabilities for a mac app package if vendor is null, empty, space or '_' */
+        if (!pkg_vendor || pkg_vendor && (*pkg_vendor == '\0' || !strncmp(pkg_vendor, " ", 1) || !strncmp(pkg_vendor, "_", 1))) {
```

Comment:
What are the cases where the pkg_vendor will be null, "", " " or "_"? because in the function `wm_vuldet_collect_agent_software` we are replacing ' ' by '_' so if I'm not wrong it would never be ' ' (space)
It just a detail, more checks are better than less checks

Reply:
I left the pre-existent checks and added some more.

For example, the space `" "` is the default unknown value of **syscollector**. But it ends up being an underscore `_` because vulnerability detector tries to convert all the spaces to match the NVD. So the vendor being a `_` it's almost an accident.

I tried to be generic to avoid a little change breaking the functionality again.